src/wallet/test/fuzz/coinselection.cpp

```diff
@@ -88,7 +88,7 @@ FUZZ_TARGET(coinselection)
     const CFeeRate effective_fee_rate{ConsumeMoney(fuzzed_data_provider, /*max=*/COIN)};
     const CAmount cost_of_change{ConsumeMoney(fuzzed_data_provider, /*max=*/COIN)};
     const CAmount min_viable_change{ConsumeMoney(fuzzed_data_provider, /*max=*/COIN)};
-    const CAmount target{fuzzed_data_provider.ConsumeIntegralInRange<CAmount>(1, MAX_MONEY)};
```

Comment:
While we don’t broadcast transactions with outputs of that amount (except OP_RETURN maybe), if it doesn’t break anywhere, a target of 1 ṩ should maybe be legal for coin selection?

IIRC, Bitcoin Core permits 294 ṩ for P2WPKH outputs, so it should at least not be bigger than that.

Reply:
I checked that 1ṩ  may break `assert(result_bnb->GetChange(coin_params.m_cost_of_change, CAmount{0}) == 0);`

Reply:
Ah, my bad, I thought this target referred to the sum of recipient outputs, but it’s actually `recipient_sum + not_input_fees`.

Is it possible that this then would need to be at least as large as the cost for the header + change output? 1000 ṩ/vB might not be enough for every feerate then.

Reply:
Nice, will leave it running tonight with some values (1000, 5000, etc) to check it as well.

Reply:
I got a crash after 7.4M attempts:

    fuzz: ../../src/wallet/test/fuzz/coinselection.cpp:120: void wallet::coinselection_fuzz_target(FuzzBufferType): Assertion `result_bnb->GetChange(coin_params.m_cost_of_change, CAmount{0}) == 0' failed.`
    
You should be able to replicate the issue with:
    
$ echo "MP3//////wT/LzMBEABL////////Wv///yWyEABLAADoQP//PP//CAAAPQAAAAAAAAAAPQEAAAAA AAAA/V12+w==" | base64 -d > crash.input 
$ FUZZ=coinselection src/test/fuzz/fuzz crash.input